code/game/objects/machinery/vending/new_marine_vendors.dm

```diff
@@ -250,7 +306,10 @@
 
 			var/type_p = L[3]
 
-			new type_p(loc)
+			var/item = new type_p(loc)
+			if(isitem(item))
```

Comment:
Might want to add this as a variable of the product, if it's meant to be placed in hands or not.

Reply:
Are there any things you can vend that aren't items?  You can only pickup items with your hands.

Reply:
There are some special effect types that spawn the crates.

Reply:
The crates are still obj/item though?

Reply:
there are `obj/effect/essentials_set/leader` that when spawned, create the crate.